source/h2_decoder.c

```diff
@@ -1159,6 +1159,42 @@ static struct aws_h2err s_flush_pseudoheaders(struct aws_h2_decoder *decoder) {
     if (has_request_pseudoheaders) {
         /* Request header-block. */
         current_block->block_type = AWS_HTTP_HEADER_BLOCK_MAIN;
+        const struct aws_string *method_string = current_block->pseudoheader_values[PSEUDOHEADER_METHOD];
+        const struct aws_string *scheme_string = current_block->pseudoheader_values[PSEUDOHEADER_SCHEME];
+        const struct aws_string *authority_string = current_block->pseudoheader_values[PSEUDOHEADER_AUTHORITY];
+        const struct aws_string *path_string = current_block->pseudoheader_values[PSEUDOHEADER_PATH];
+
+        if (!method_string) {
+            DECODER_LOG(ERROR, decoder, "Request is missing :method.");
+            goto malformed;
+        }
+        if (!aws_strutil_is_http_token(aws_byte_cursor_from_string(method_string))) {
+            DECODER_LOG(ERROR, decoder, "Request method is invalid.");
+            DECODER_LOGF(
+                DEBUG,
+                decoder,
+                "Bad method is: '" PRInSTR "'",
+                AWS_BYTE_CURSOR_PRI(aws_byte_cursor_from_string(method_string)));
+            goto malformed;
+        }
+        if (aws_string_eq_c_str(method_string, "CONNECT")) {
+            if (scheme_string || path_string) {
+                /* RFC-9113 8.5 The ":scheme" and ":path" pseudo-header fields MUST be omitted for CONNECT method */
+                DECODER_LOG(ERROR, decoder, "CONNECT request has :path or :scheme that are must omitted.");
```

Comment:
do we need to check that these things aren't empty strings?

I see you checking the request_string, but not the others

Reply:
It was checked before. from the `s_process_header_field` at [here](https://github.com/awslabs/aws-c-http/pull/383/files#diff-ade185b92b6e3655280095a0215637f556a63fd1e236f60a46663b8e96605046R1291)